substrate/network/src/sync.rs

```diff
@@ -65,16 +65,18 @@ pub enum SyncState {
 
 /// Syncing status and statistics
 #[derive(Clone)]
-pub struct Status {
+pub struct Status<B: BlockT> {
 	/// Current global sync state.
 	pub state: SyncState,
 	/// Target sync block number.
-	pub best_seen_block: Option<BlockNumber>,
+	pub best_seen_block: Option<<B::Header as HeaderT>::Number>,
 }
 
-impl ChainSync {
+impl<B: BlockT> ChainSync<B> where
+	B::Header: HeaderT<Number=u64>,
```

Comment:
Why the `u64` requirement?

Reply:
for now, it maintains the same properties as the current sync code. it would be easy to change the sync logic to support more block number types in a follow-up PR, but since this one was getting giant I thought it better to leave that out for now.